### PR TITLE
Implement `spanOf` method to compute the`SourceLocation` that encloses one or more others.

### PR DESCRIPTION
**Pull request by aaronhurst-google**
Implement `unionOf` method to compute the containing range of one or more `SourceLocation` values

Add unit tests of several cases: null, reflexivity, idempotency, others.

This enhancement is to support PMD nodes that map to multiple Summit nodes: their source intervals will be the union of the Summit nodes.

### CODE REVIEW

File: src/main/java/com/google/summit/ast/SourceLocation.kt [hunk 1]
```diff
@@ -66,3 +66,38 @@ data class SourceLocation(
...
+    x.endLine > y.endLine -> x
+    y.endLine > x.endLine -> y
+    x.endColumn == null -> y
+    y.endColumn == null -> x
+    x.endColumn > y.endColumn -> x
+    else -> y
+  }
+
+/** Unions one or more [SourceLocation] ranges to the containing range. */
+fun unionOf(vararg locs: SourceLocation): SourceLocation {
```
Comment (zorzella) [inline src/main/java/com/google/summit/ast/SourceLocation.kt]: Both doc and method name "union" is a bit misleading (or ambiguous), as may imply the mathematical concept of union (which would not include anything "between", as it happens here). That said, I'm struggling to think of a better name...

At least maybe document this is not the math concept of union, and that it includes everything in between the sections.

**Review by zorzella — APPROVED** [review]

File: src/main/java/com/google/summit/ast/SourceLocation.kt (hunk 1 above)
Comment (aaronhurst-google) [reply src/main/java/com/google/summit/ast/SourceLocation.kt]: I changed it to `spanOf`
